bip-0360.mediawiki

```diff
@@ -11,696 +11,397 @@
 </source>
 
-P2TR key path spend witness (66 bytes):
+P2TR key-path spend witness (66 bytes):
+
 <source>
 [count] (1 byte), # Number of elements in the witness
 [size] signature (1 + 64 bytes = 65 bytes)
 </source>
 
-Thus, the P2QRH input would be 103 - 66 = 37 bytes larger than a P2TR key path spend input.
+Thus, the P2TSH input would be 103 - 66 = 37 bytes larger than a P2TR key path spend input.
+
+If the Merkle tree has more than a single leaf, then the Merkle path must be included in the control block, increasing the size by ''32 * m'' bytes. This would make such input 37 + 32 * m bytes larger than a P2TR key path spend input.<ref>If ''m >= 8'', then the compact size will use 3 bytes rather than 1 byte</ref>
+
+P2TSH witness ''(103 + 32*m bytes)'':
 
-If the Merkle tree has more than a single leaf, then the Merkle path must be included in
-the control block.
-P2QRH witness (103+32*m bytes)
 <source>
 [count] (1 byte), # Number of elements in the witness
-[size] signature  (64 + 1 bytes = 65 bytes),
-tapleaf script = [size] [OP_PUSHBYTES_32, 32-byte public key, OP_CHECKSIG] (34 + 1 bytes = 35 bytes),
-control block = [size] [control byte, 32 * m byte Merkle path]  (1 + 1 + 32 * m = 2 + 32 * m bytes)
+[size] signature (64 + 1 bytes = 65 bytes),
+tapleaf script = [size] [OP_PUSHBYTES_32, 32-byte scriptPubKey, OP_CHECKSIG] (34 + 1 bytes = 35 bytes),
+control block = [size] [control byte] [Merkle path] (1 + 1 + 32*m = 2 + 32*m bytes)
 </source>
 
-For a Merkle path of length m, it would add an additional ~32 * m bytes to the P2QRH input. This would
-make it 37 + 32 * m bytes larger than a P2TR key path spend input<ref>If m >= 8, then the compact size will use 3 bytes rather than 1 byte</ref>.
+====Comparison with P2TR script path spend====
 
-Considering a P2QRH output that has a PQ signature leaf and a Schnorr leaf. The P2QRH witness to spend the Schnorr path
-would be 103 + 32 * 1 = 135 bytes. It is unfortunate that we can not use the key path spend optimization for P2QRH inputs, but the key path spend optimization is
-exactly what makes P2TR vulnerable to quantum attacks. If key path spend was quantum resistant we wouldn't need P2QRH at all.
+A P2TSH input will be smaller than an equivalent script path spend for P2TR inputs. This is because P2TSH inputs do not require inclusion of a public key in the control block to open the commitment to the Merkle root. A P2TSH input will be 32 bytes smaller than an equivalent P2TR script path spend input.
 
-'''P2TR script path spend''' P2QRH inputs will be smaller than equivalent script path spend P2TR inputs. This is because P2QRH inputs
-do not require that the input includes a public key in the control block to open the commitment to the Merkle root.
-An equivalent P2QRH input will be 32 bytes smaller than a P2TR script path spend input.
+====Consider a P2TSH output with a post-quantum signature leaf and a Schnorr leaf====
```

Comment:
Should we be capitalizing post-quantum? No, right?

Reply:
There is only one place where we capitalize it, and that's because it's a title